commit 001e4e5537d83d5b5814a642a47d7691de80a66f
Author: Peter Oskolkov <posk@google.com>
Date:   Mon Sep 17 10:20:53 2018 -0700

    net/ipv6: do not copy dst flags on rt init
    
    [ Upstream commit 30bfd93062814d6767e452a8f5ddcd97f7e38c7e ]
    
    DST_NOCOUNT in dst_entry::flags tracks whether the entry counts
    toward route cache size (net->ipv6.sysctl.ip6_rt_max_size).
    
    If the flag is NOT set, dst_ops::pcpuc_entries counter is incremented
    in dist_init() and decremented in dst_destroy().
    
    This flag is tied to allocation/deallocation of dst_entry and
    should not be copied from another dst/route. Otherwise it can happen
    that dst_ops::pcpuc_entries counter grows until no new routes can
    be allocated because the counter reached ip6_rt_max_size due to
    DST_NOCOUNT not set and thus no counter decrements on gc-ed routes.
    
    Fixes: 3b6761d18bc1 ("net/ipv6: Move dst flags to booleans in fib entries")
    Cc: David Ahern <dsahern@gmail.com>
    Acked-by: Wei Wang <weiwan@google.com>
    Signed-off-by: Peter Oskolkov <posk@google.com>
    Reviewed-by: David Ahern <dsahern@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/ipv6/route.c b/net/ipv6/route.c
index 3eed045c65a5..480a79f47c52 100644
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -946,8 +946,6 @@ static void ip6_rt_init_dst_reject(struct rt6_info *rt, struct fib6_info *ort)
 
 static void ip6_rt_init_dst(struct rt6_info *rt, struct fib6_info *ort)
 {
-	rt->dst.flags |= fib6_info_dst_flags(ort);
-
 	if (ort->fib6_flags & RTF_REJECT) {
 		ip6_rt_init_dst_reject(rt, ort);
 		return;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1019_001e4e5537d83d5b5814a642a47d7691de80a66f.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1019_001e4e5537d83d5b5814a642a47d7691de80a66f.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
